### PR TITLE
Error Logging for ID Uploader

### PR DESCRIPTION
**Pull request by isaacbleecker**
Added generic error logging service to UI. Includes method to get user's browser and OS. Currently only used by ID Uploader, to record successes and failures, as well as the browser / os.
It uses a new logging route in the API, to record logs to the database.

Hopefully, this logging service will be helpful for collecting errors from actual website usage. 

Currently the size of the error message seems to be limited -- might need to look into sending multipart file content, or something that can handle larger messages. We were unable to send an entire 6 mb blob, but for typical strings it is currently fine.

### CODE REVIEW

File: src/services/errorLog.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  };
+  return http.post('log/add', data);
+};
+
+const postErrorMessage = message => {
+  return http.post('log', message);
+};
+
+//Code modified from https://medium.com/creative-technology-concepts-code/detect-device-browser-and-version-using-javascript-8b511906745
+function matchItem(string, data) {
```
Comment (joshuaRogers24) [inline src/services/errorLog.js]: can you use the React function declaration syntax here, as you did on lines 9 and 18?

File: src/services/errorLog.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      };
+    }
+  }
+  return { name: 'unknown' };
+}
+
+const parseNavigator = navigator => {
+  //Code modified from https://medium.com/creative-technology-concepts-code/detect-device-browser-and-version-using-javascript-8b511906745
+  var oses = [
+    { name: 'Windows Phone', value: 'Windows Phone' },
```
Comment (joshuaRogers24) [inline src/services/errorLog.js]: is oses plural for os? Is there a clearer name for this array?

File: src/services/membership.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ import http from './http';
  * @return {Promise<any>} Response
  */
 function addMembership(data) {
-  return http.post(`memberships`, data).catch(reason => {
+  return http.post('memberships', data).catch(reason => {
     console.log(reason);
```
Comment (joshuaRogers24) [inline src/services/membership.js]: was this just a "bunny-trail" change, irrelevant to the error log?

File: src/views/IDUploader/index.js [hunk 4]
```diff
@@ -57,17 +57,20 @@ export default class IDUploader extends Component {
   };
 
   async postCroppedImage(croppedImage, n) {
-    console.log('ID photo resubmission #' + n);
+    let logMessage = 'ID photo submission #' + n + ' from ' + errorLog.parseNavigator(navigator);
```
Comment (joshuaRogers24) [inline src/views/IDUploader/index.js]: is there a more descriptive name for the 'n' parameter?

Comment (joshuaRogers24) [inline src/views/IDUploader/index.js]: Is 'n' clear enough for this parameter's name?

File: src/services/membership.js (hunk 3 above)
Comment (isaacbleecker) [reply src/services/membership.js]: Yes, this one was just a change for consistency, as the special back tick characters only need to be used when there is a multi-line string or a variable/expression inside the string.
It's not relevant to error logging.

File: src/services/errorLog.js (hunk 2 above)
Comment (isaacbleecker) [reply src/services/errorLog.js]: That was the intent. Perhaps just naming it "os" would be clearer? Or I could use "OperatingSystems" ?

**Pull request merged**